src/consensus.rs

```diff
@@ -43,15 +43,15 @@ pub(crate) async fn run<H: Hasher + 'static>(
 
     // send a new parent candidate to the creator
     terminal.register_post_insert_hook(Box::new(move |u| {
-        if let Err(e) = parents_tx.unbounded_send(u.into()) {
+        parents_tx.unbounded_send(u.into()).map_err(|e| {
```

Comment:
I'm not sure what the current consensus is – are we going to use `expect` for all channels internal to AlephBFT or not? Or was it some other criterion?

Reply:
That would be preferred but here we cannot as potential panic would kill just the terminal task and all the rest components wouldn't know about that. The change tries to propagate this info to all the components.